src/main/scala/rx/lang/scala/observables/AsyncOnSubscribe.scala

```diff
@@ -0,0 +1,120 @@
+package rx.lang.scala.observables
+
+import rx.annotations.Experimental
+import rx.lang.scala.{Notification, Observable, Observer}
+
+/**
+  * An utility class to create observables that start acting when subscribed to and respond correctly
+  * to back pressure requests from subscribers.
+  */
+object AsyncOnSubscribe {
+  /**
+    * Generates an `Observable` that synchronously calls the provided `next` function
```

Comment:
The `next` is still synchronously called (see [RxJava javadoc](https://github.com/ReactiveX/RxJava/blob/dc63e5350d08f08b4c1229df812f4a0a19f1083e/src/main/java/rx/observables/AsyncOnSubscribe.java#L159)). The async comes from `next`'s resultant `Observable[T]`, which can asynchronously deliver up to `requested` items.

I'll see if I can improve the docs to make this more clear.